### PR TITLE
fix: deposit `Claimed` state waits for e-cash to be in wallet

### PR DESCRIPTION
**Pull request by elsirion**
Follow-up to #5915.

### CODE REVIEW

File: modules/fedimint-wallet-tests/tests/tests.rs [hunk 1]
```diff
@@ -192,6 +192,10 @@ async fn on_chain_peg_in_and_peg_out_happy_case() -> anyhow::Result<()> {
         deposit_updates.next().await.unwrap(),
         DepositStateV2::WaitingForConfirmation { .. }
     ));
+    assert!(matches!(
+        deposit_updates.next().await.unwrap(),
+        DepositStateV2::Confirmed { .. }
+    ));
```
Comment (shaurya947) [inline modules/fedimint-wallet-tests/tests/tests.rs:198]: nit: would be nice if we can test that after `Claimed` state is emitted, the balance is correct (which would verify that all the e-cash has been issued)

**Review by shaurya947 — APPROVED** [review]

File: modules/fedimint-wallet-tests/tests/tests.rs (hunk 1 above)
Comment (elsirion) [reply modules/fedimint-wallet-tests/tests/tests.rs:238]: I added a commit to that end

File: modules/fedimint-wallet-tests/tests/tests.rs [hunk 2]
```diff
@@ -188,14 +197,57 @@ async fn on_chain_peg_in_and_peg_out_happy_case() -> anyhow::Result<()> {
...
+    info!("Waiting for confirmation");
     assert!(matches!(
         deposit_updates.next().await.unwrap(),
-        DepositStateV2::WaitingForConfirmation { .. }
+        DepositStateV2::WaitingForConfirmation { btc_out_point, .. } if btc_out_point.txid == tx.txid()
     ));
+
+    bitcoin.mine_blocks(finality_delay).await;
+
+    // Afaik technically not necessary, but useful to speed up test (should probably
```
Comment (bradleystachurski) [inline modules/fedimint-wallet-tests/tests/tests.rs:218]: Hmm, I wouldn't have expected this to be necessary since we have a few checks if we're in a test and poll aggressively (1ms or 100ms depending on path).

I'm a bit suspicious of the behavior outside of a test environment now.

**Review by bradleystachurski — APPROVED** [review]
Should we backport and create another rc?

File: modules/fedimint-wallet-tests/tests/tests.rs (hunk 2 above)
Comment (bradleystachurski) [reply modules/fedimint-wallet-tests/tests/tests.rs:218]: Interesting, looks like `is_running_in_test_env` returns false when running locally with `cargo test -p fedimint-wallet-tests`

Comment (elsirion) [reply modules/fedimint-wallet-tests/tests/tests.rs:218]: I read through the code that determines wake up times and it's wayyyy over-engineered imo. There are two layers of wake up logic:

1. The outer one is told by the lowest of all `next_check_time` when to wake up next. There is also a `min_sleep` that responds to being in a test env, but that only sets a minimum wait before anything can happen, so doesn't help.
2. The inner layer is triggered by the outer and determines which addresses actually need checking. This one checks more often if it runs in the dev env, but the outer layer blocks us from getting there.

I think the logic could be massively simplified, it's not that hard of a problem and the complexity here seems accidental.

**Comment by fedimint-backports** [comment]
Backport failed for `releases/v0.4`, because it was unable to cherry-pick the commit(s).

Please cherry-pick the changes locally and resolve any conflicts.
```bash
git fetch origin releases/v0.4
git worktree add -d .worktree/backport-5933-to-releases/v0.4 origin/releases/v0.4
cd .worktree/backport-5933-to-releases/v0.4
git switch --create backport-5933-to-releases/v0.4
git cherry-pick -x 5f0eedb4ce5387976f3f768e37ab4dc1a09a0dbf 9b65bbd7e638f2b93d0b32447270c4320c4c915c
```
